packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -2179,6 +2264,7 @@ class ThemeData with Diagnosticable {
     properties.add(EnumProperty<AndroidOverscrollIndicator>('androidOverscrollIndicator', androidOverscrollIndicator, defaultValue: null, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<bool>('applyElevationOverlayColor', applyElevationOverlayColor, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<NoDefaultCupertinoThemeData>('cupertinoOverrideTheme', cupertinoOverrideTheme, defaultValue: defaultData.cupertinoOverrideTheme, level: DiagnosticLevel.debug));
+    properties.add(DiagnosticsProperty<Map<Object, ThemeExtension<dynamic>>>('extensions', extensions, defaultValue: defaultData.extensions, level: DiagnosticLevel.debug));
```

Comment:
For cleaner logic, maybe 
IterableProperty<ThemeExtension<Objet>>('extensions', extensions.values(), ...);

On similar note, maybe worth making ThemeExtension implements Diagnosticable